### PR TITLE
Minor enhancements to the fixtures

### PR DESCRIPTION
**Pull request by echarles**
While reading the tests, I found a few minor enhancements like removing the param from base_url fixture and adding doc to app fixture.\

I may push in the coming day more minore udpates.

My understanding is that only the fixtures defined in `pytest_plugin.py` are usable in other projects (like nbclassic). Then, how do we decide which fixture to expose? e.g. I can see a use of `ws_fetch` defined in `test_api.py` in kernel related projects but today this is not possible.

### CODE REVIEW

**Comment by Zsailer** [comment]
> I can see a use of ws_fetch defined in test_api.py in kernel related projects but today this is not possible.

I totally agree. In fact, I think #160 moves this fixture into the base level conftest (before I made the plugin). I think it would be work moving ws_fetch into the plugin.

**Comment by echarles** [comment]
> I totally agree. In fact, I think #160 moves this fixture into the base level conftest (before I made the plugin). I think it would be work moving ws_fetch into the plugin.

True, it moves it [here](https://github.com/jupyter/jupyter_server/pull/160/files#diff-484462fced51d1a06b1d93b4a44dd535R202). I will pick that change in this PR and will go through the tests to identify other candidates.

**Comment by echarles** [comment]
I have moved `ws_fetch`, `kernelspecs` and `contents_manager` fixtures to pytest_plugin for 3rd party consumption. WDYT?

File: jupyter_server/pytest_plugin.py [hunk 1]
```diff
@@ -62,6 +63,11 @@ def mkdir(tmp_path, *parts):
 env_config_path = pytest.fixture(
     lambda tmp_path: mkdir(tmp_path, "env", "etc", "jupyter")
 )
+some_resource = u"The very model of a modern major general"
```
Comment (Zsailer) [inline jupyter_server/pytest_plugin.py]: Nice Hamilton quote. 😉

File: jupyter_server/pytest_plugin.py [hunk 2]
```diff
@@ -198,6 +205,52 @@ def client_fetch(*parts, headers={}, params={}, **kwargs):
...
+        sample_kernel_dir.mkdir(parents=True)
+        # Create kernel json file
+        sample_kernel_file = sample_kernel_dir.joinpath('kernel.json')
+        sample_kernel_file.write_text(json.dumps(sample_kernel_json))
+        # Create resources text
+        sample_kernel_resources = sample_kernel_dir.joinpath('resource.txt')
+        sample_kernel_resources.write_text(some_resource)
+
+
+# contents_manager_atomic = pytest.fixture(lambda tmp_path: FileContentsManager(root_dir=str(tmp_path), use_atomic_writing=True))
```
Comment (Zsailer) [inline jupyter_server/pytest_plugin.py]: Do you want to keep these lines that are commented out?

File: jupyter_server/pytest_plugin.py (hunk 1 above)
Comment (echarles) [reply jupyter_server/pytest_plugin.py]: Not sure here... I have merge master and they are still there... I can only say it is not me as I do not know Hamilton...

File: jupyter_server/pytest_plugin.py (hunk 2 above)
Comment (echarles) [reply jupyter_server/pytest_plugin.py]: commented lines are now removed.

**Comment by Zsailer** [comment]
LGTM! Thanks, @echarles!

**Pull request merged**